### DISCUSSION TITLE
Scaling content to fill a container

### DISCUSSION THREAD

**Discussion by hcsch** [typst/typst#1256] [Q&A] [answered]
Is there currently a way to scale content (e.g. text) to fill a container either vertically or horizontally?

For example I want to scale a title to fit exactly into the width of one column, i.e. I want its width to match the available line width for text in that column.

I haven't found any way to measure line width, column width, or to scale arbitrary content to some fraction of the containers width. Measuring a box with `width: 100%` just gets me a width of 0, which so far was the only thing I could think of that could possibly have worked..

The following sort-of works if you already know the exact width you want to scale to ~~, but the (layout affecting) height of the resulting box is still that of the original text, from what I can tell (i.e. the height parameter of the box is just ignored)~~ _Edit: I think I was making bad assumptions here, using `bottom-edge: descender` in `text` is what I wanted in this case_:
```typst
#let min-width-box(target-width, body) = style(styles => {
    let default-body-size = measure(body, styles)

    if target-width > default-body-size.width {
        let scaling-factor = (target-width / default-body-size.width);
        box(
            width: target-width,
            height: default-body-size.height * scaling-factor,
            scale(x: scaling-factor * 100%, y: scaling-factor * 100%, origin: top+left, body)
        )
    } else {
        box(body)
    }
})

#set page(width: 10cm, height: 5cm, margin: 1cm)

#place(rect(width: 100%, height: 100%, fill: purple))
#min-width-box(10cm - 2*1cm, text(1pt, weight: "extrabold", fill: white, [Sample Text]))
#lorem(20)
```
![image](https://github.com/typst/typst/assets/34245439/83c8559b-72f6-43a1-b91b-994c7aa700e4)

**Comment by laurmaedje** [comment, marked as answer]
This might help: https://typst.app/docs/reference/meta/layout/

**Reply by hcsch** [reply]
That does indeed help, thanks a lot. I'm not sure how I missed that function. Does the docs search currently work for the content of docs pages as well? Searching for "container size" doesn't get me this function alas (though searching `site:typst.app/docs container size` on DDG does get me there)

**Reply by laurmaedje** [reply]
It does search the contents, but the search is not particularly advanced. It splits up your query into individual words, finds all pages that contain the words and then ranks the matches by the title (not the content). As such, all other usages of "container" rank equally well although the don't continue with "size".

**Reply by hcsch** [reply]
Ah ok, I suppose I'll be using an external search engine a bit more for now then. Thanks for helping me out ^^
